projects/xy-finance/index.js

```diff
@@ -0,0 +1,48 @@
+const utils = require('../helper/utils')
+
+const chains = {
+  1: 'ethereum',
+  10: 'optimism',
+  25: 'cronos',
+  56: 'bsc',
+  108: 'thundercore',
+  137: 'polygon',
+  250: 'fantom',
+  321: 'kcc',
+  592: 'astar',
+  1285: 'moonriver',
+  42161: 'arbitrum',
+  43114: 'avax',
+}
+
+const url = 'https://api.xy.finance/stats/v1/tvl'
+
+function fetchChain(chain) {
+  return async () => {
+    let req_url = chain === null ? url : url + '?chainId=' + chain
+    const { data } = await utils.fetchURL(req_url)
+    const tokensInChain = chain === null ? data.bridgeTokenList : data.bridgeTokenList.filter(p => p.chainId === chain)
+    let total = 0
+    tokensInChain.forEach((item) => {
+      const tvl = Number(item.tvl || 0)
+      total += tvl
+    })
+    return total
+  }
+}
+
+const chainTvls = {}
+Object.keys(chains).forEach((chain) => {
+  const chainName = chains[chain]
+  chainTvls[chainName === 'avax' ? 'avalanche' : chainName] = {
+    fetch: fetchChain(Number(chain))
+  }
+})
+
+module.exports = {
+  misrepresentedTokens: true,
```

Comment:
他們有文件列出我們需要（or optionally）export 什麼嗎？有點好奇 `misrepresentedTokens` 和 `timetravel` 用來做啥的。

Reply:
下面是他們 export 的定義。我們沒有過去的資料，所以 `timetravel` 為 `false`，`misrepresentedTokens` 這個我不太確定他們的意思，但我們應該沒有遵守任何 token 使用的規範，所以就設為 `true`，然後各鏈的 TVL 就各自輸出成 anonymous function
```
The module exports must be constructed correctly, and use the correct keys, so that the DefiLlama UI can show
 your data. Nest chain TVL (and separate types of TVL like staking, pool2 etc) inside the chain key (eg 'bsc', 'ethereum').

Please also let us know:

timetravel (bool) - if we can backfill data with your adapter. 
Most SDK adapters will allow this, but not all. For example, if you fetch a list of live contracts from an API before
querying data on-chain, timetravel should be 'false'.

misrepresentedTokens (bool) - if you have used token substitutions at any point in the adapter this should
be 'true'.

methodology (string) - this is a small description that will explain to DefiLlama users how the adapter
works out your protocol's TVL.
```

Reply:
感謝資訊